src/url.rs

```diff
@@ -7,18 +7,74 @@ const QUOTES: &[char] = &['\'', '\"'];
 /// Scan for URLs starting from the trigger character ":", requires "://".
 ///
 /// Based on RFC 3986.
-pub struct UrlScanner {}
+pub struct UrlScanner;
 
 impl Scanner for UrlScanner {
-    fn scan(&self, s: &str, colon: usize) -> Option<Range<usize>> {
-        let after_slash_slash = colon + 3;
+    /// Scan for an URL at the given separator index in the string.
+    ///
+    /// The kind of separator that was used (`://` vs `.`) has effect on whether URLs with no
+    /// schemes are found.
+    ///
+    /// Returns `None` if none was found, or if an invalid separator index was given.
+    fn scan(&self, s: &str, separator: usize) -> Option<Range<usize>> {
+        // There must be something before separator for scheme or host
+        if separator == 0 {
+            return None;
+        }
+
+        // Detect used separator, being `://` or `.`
+        let (is_slash_slash, separator_len) = if s[separator..].starts_with("://") {
+            (true, "://".len())
+        } else if s[separator..].starts_with('.') {
+            (false, ".".len())
+        } else {
+            return None;
+        };
+        let after_separator = separator + separator_len;
+
+        // When scanning a dot, for URLs without a scheme, do some addadditional checks
```

Comment:
typo: addadditional -> additional

Reply:
Fixed. Will wait with pushing until merging to prevent invalidating these review comments.